cardano-constitution/src/Cardano/Constitution/Config/Predicate/Types.hs

```diff
@@ -0,0 +1,52 @@
+{-# LANGUAGE DeriveAnyClass  #-}
+{-# LANGUAGE TemplateHaskell #-}
+module Cardano.Constitution.Config.Predicate.Types
+    ( PredName (..)
+    , PredValue
+    , PredMeaning
+    , PredMeaningApplied
+    ) where
+
+import PlutusPrelude (lowerInitialChar)
+import PlutusTx as Tx (makeLift)
+import PlutusTx.Eq qualified as Tx
+
+import Data.Aeson as Aeson
+import GHC.Generics
+import Language.Haskell.TH.Syntax as TH (Lift)
+import Prelude qualified as Haskell
+
+-- | Unresolved predicate name, as read from JSON
+data PredName =
+    MinValue
+  | MaxValue
+  deriving stock (Haskell.Eq, Haskell.Ord, Haskell.Show, TH.Lift, Generic)
+  deriving anyclass (Aeson.FromJSON)
+
+-- | Unresolved predicate value, as read from JSON
+type PredValue = Haskell.Integer
+
+instance Tx.Eq PredName where
+    {-# INLINABLE (==) #-}
+    MinValue == MinValue = Haskell.True
+    MaxValue == MaxValue = Haskell.True
+    _ == _               = Haskell.False
```

Comment:
Please explicitly enumerate all the possibilities of the first argument, so that we'll get a warning here when a new predicate is added. We do it for `Eq` sometimes, e.g.

```haskell
eqTermM Constant{} _ = empty
eqTermM Builtin{}  _ = empty
eqTermM Var{}      _ = empty
eqTermM LamAbs{}   _ = empty
eqTermM Apply{}    _ = empty
eqTermM Delay{}    _ = empty
eqTermM Force{}    _ = empty
eqTermM Error{}    _ = empty
eqTermM Constr{}    _ = empty
eqTermM Case{}    _ = empty
```